sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -36,29 +36,41 @@ import org.apache.spark.sql.internal.StaticSQLConf._
 import org.apache.spark.status.ElementTrackingStore
 import org.apache.spark.util.{MutableURLClassLoader, Utils}
 
-
 /**
  * A class that holds all state shared across sessions in a given [[SQLContext]].
  */
-private[sql] class SharedState(val sparkContext: SparkContext) extends Logging {
+private[sql] class SharedState(val sparkContext: SparkContext, initConfig: Map[String, String])
+  extends Logging {
+  private val conf = sparkContext.conf.clone()
+  private val hadoopConf = new Configuration(sparkContext.hadoopConfiguration)
+
+  // If `SparkSession` is instantiated using an existing `SparkContext` instance and no existing
+  // `SharedState`, all `SparkSession` level configurations have higher priority to generate a
+  // `SharedState` instance. This will be done only once then shared across `SparkSession`s
+  initConfig.foreach { case (k, v) =>
+    logDebug(s"Applying initiate SparkSession options to SparkConf/HadoopConf: $k -> $v")
+    conf.set(k, v)
+    hadoopConf.set(k, v)
+  }
+  logInfo("Applied all initiate SparkSession options to the brand new SharedState")
```

Comment:
initiate -> initial

Reply:
BTW do we need this log? We already have a log for each config.

Reply:
debug level for each

Reply:
I don't see how useful this log is, it basically means a `SessionState` is created.